libvast/src/system/spawn_index.cpp

```diff
@@ -32,11 +32,12 @@ maybe_actor spawn_index(node_actor* self, spawn_arguments& args) {
     return get_or(args.invocation.options, key, default_value);
   };
   namespace sd = vast::defaults::system;
-  auto result = self->spawn(index, args.dir / args.label,
-                            opt("max-events", sd::max_partition_size),
-                            opt("max-parts", sd::max_in_mem_partitions),
-                            opt("taste-parts", sd::taste_partitions),
-                            opt("max_queries", sd::num_query_supervisors));
+  auto result = self->spawn(
+    index, args.dir / args.label,
+    opt("system.max-partition-size", sd::max_partition_size),
+    opt("system.max-resident-partitions", sd::max_in_mem_partitions),
+    opt("system.max-taste-partitions", sd::taste_partitions),
+    opt("system.max-queries", sd::num_query_supervisors));
```

Comment:
This might be my misunderstanding of the option processing, and maybe @dominiklohmann can fill me in, but why don't we have hierarchical options here, such as `system.index.max-partition-size`?

Reply:
They should be `spawn.index.max-partition-size` to correspond to the `vast spawn index` command.

Reply:
Okay. @tobim mind adjusting this accordingly?

Reply:
I think we should talk about this first. As is right now, nobody is using the `vast spawn index` command directly. Never.

It supports these options (also available via `vast spawn index help`):

```cpp
spawn->add_subcommand(
  "index", "creates a new index", "",
  opts()
    .add<size_t>("max-events,e", "maximum events per partition")
    .add<size_t>("max-parts,p", "maximum number of in-memory partitions")
    .add<size_t>("taste-parts,t", "number of immediately scheduled "
                                  "partitions")
    .add<size_t>("max-queries,q", "maximum number of concurrent queries"));
```

As nobody is ever invoking these commands directly, they _might_ actually make more sense in the `system` category, which is what we use for options that are placed immediately after the root command.

Reply:
Following our discussion in Slack, these should just stay in the `system` category for now. Resolving this.

Reply:
I changed these options in one of the commits in here to `system.*` (with more expressive names too).

The problem with `system.index.max-partition-size` or `spawn.index.max-partition-size` is that the command line parser does not accept fully qualified options at the moment. Maybe it is desirable to change that.